lightning/src/util/events.rs

```diff
@@ -180,10 +180,14 @@ pub enum Event {
 	/// [`ChannelManager::fail_htlc_backwards`] within the HTLC's timeout, the HTLC will be
 	/// automatically failed.
 	///
+	/// # Note
+	/// In rare cases, multiple `PaymentReceived` events may be generated for the same payment.
```

Comment:
I wouldn't call this rare? I mean if someone sends a payment twice for the same payment, we'll happily generate this twice.

Reply:
True, see 7503d196c971475b3001e80a2a7a4ec74b62ed39